python/meterpreter/meterpreter.py

```diff
@@ -1258,11 +1261,13 @@ def add_channel(self, channel):
         self.next_channel_id += 1
         return idx
 
-    def add_process(self, process):
+    def add_process(self, process, win32_handle = None):
         idx = self.next_process_id
-        self.processes[idx] = process
+        self.processes[idx] = (process, win32_handle)
         debug_print('[*] added process id: ' + str(idx))
         self.next_process_id += 1
+        if win32_handle:
+            idx = win32_handle
```

Comment:
If the handle is returned then it should also be the key for `processes`. That is to say if `win32_handle` is present it should be used as `idx` instead of the next_process_id. The value returned from here is actually supposed to be process handle in Meterpreter terms. If we're on Windows, this is probably where we actually want to call `OpenProcess` if we have windll available.

```suggestion
    def add_process(self, process):
        if has_windll:
            k32 = ctypes.windll.kernel32
            # still need to update OpenProcess here
            handle = k32.OpenProcess(PROCESS_ALL_ACCESS, False, proc_h.pid)
        else:
            handle = self.next_process_id
            self.next_process_id += 1
        self.processes[handle] = process
        debug_print('[*] added process id: ' + str(proc_h.pid) + ', handle: ' + str(handle))
        return handle
```